### PR TITLE
Remove makefiles from docs building

### PR DESCRIPTION
**Pull request by ssbarnea**
- Remove makefiles
- Updates docs contributing readme to match our setup
- Updates html output directory to make it compatible with zuul

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/578

### CODE REVIEW

File: docs/docsite/README.md [hunk 1]
```diff
@@ -1,28 +1,9 @@
 # Documentation source for Ansible Lint
 
-Contains the source behind [docs.ansible.com/ansible-lint](https://docs.ansible.com/ansible-lint)
+To build the docs, run `tox -e docs`. At the end of the build, you will
+see the local location of your builded docs.
```
Comment (webknjaz) [inline docs/docsite/README.md:4]:
```suggestion
see the local location of your built docs.
```

File: docs/docsite/README.md [hunk 2]
```diff
@@ -1,28 +1,9 @@
...
-# Within the Galaxy project tree, set your working directory to docs/docsite
-$ cd docs/docsite
-
-# Build a local copy of the static files
-$ make webdocs
-
-# Open the site in your default browser
-$ xdg-open _build/html/index.html
-```
+Building docs locally is not identical with CI/CD builds, so you
```
Comment (webknjaz) [inline docs/docsite/README.md:6]:
```suggestion
Building docs locally may not be identical with CI/CD builds, so you
```

File: tox.ini [hunk 3]
```diff
@@ -103,9 +103,22 @@ passenv =
...
+description = Invoke sphinx-build to build the HTML docs
+basepython = python3
+deps = -r{toxinidir}/docs/requirements.txt
+passenv = *
+usedevelop = False
+commands =
+    python -m sphinx \
+      -a -n -W \
+      -b html --color \
+      -d "{toxworkdir}/.docstree" \
```
Comment (webknjaz) [inline tox.ini:115]: this should probably be
```suggestion
      -d "{temp_dir}/.docstree" \
```

File: tox.ini [hunk 4]
```diff
@@ -103,9 +103,22 @@ passenv =
...
+      docs/docsite/rst/ "{toxinidir}/.tox/docs/html"
```
Comment (webknjaz) [inline tox.ini:116]: Instead, I'd use `change_dir`.  And also env dir.
```suggestion
      rst/ \ 
      "{envdir}/html"
```

File: tox.ini [hunk 5]
```diff
@@ -103,9 +103,22 @@ passenv =
...
 [testenv:docs]
-whitelist_externals = make
-deps = sphinx
-commands = make -C docs/docsite htmldocs
+description = Invoke sphinx-build to build the HTML docs
+basepython = python3
+deps = -r{toxinidir}/docs/requirements.txt
+passenv = *
+usedevelop = False
+commands =
```
Comment (webknjaz) [inline tox.ini:111]:
```suggestion
change_dir = docs/docsite/
commands =
```

File: tox.ini [hunk 6]
```diff
@@ -103,9 +103,22 @@ passenv =
   PRE_COMMIT_HOME
 
 [testenv:docs]
-whitelist_externals = make
-deps = sphinx
-commands = make -C docs/docsite htmldocs
+description = Invoke sphinx-build to build the HTML docs
+basepython = python3
+deps = -r{toxinidir}/docs/requirements.txt
```
Comment (webknjaz) [inline tox.ini:108]: depends on https://github.com/ansible/ansible-lint/pull/861

File: tox.ini [hunk 7]
```diff
@@ -103,9 +103,22 @@ passenv =
...
+passenv = *
```
Comment (webknjaz) [inline tox.ini:109]: this is unnecessary

File: tox.ini [hunk 8]
```diff
@@ -103,9 +103,22 @@ passenv =
...
+usedevelop = False
+commands =
+    python -m sphinx \
```
Comment (webknjaz) [inline tox.ini:112]:
```suggestion
    {envpython} -m sphinx \
```

File: tox.ini [hunk 9]
```diff
@@ -103,9 +103,22 @@ passenv =
...
+      -a -n -W \
+      -b html --color \
+      -d "{toxworkdir}/.docstree" \
+      docs/docsite/rst/ "{toxinidir}/.tox/docs/html"
+
+    # Print out the output docs dir and a way to serve html:
+    python -c \
+    'import pathlib; '\
+    'docs_dir = pathlib.Path(r"{toxinidir}") / ".tox/docs/html"; index_file = docs_dir / "index.html"; print(f"\nDocumentation available at file://\{index_file\}\n\nTo serve docs, run: python3 -m http.server --directory \{docs_dir\} 0\n")'
```
Comment (webknjaz) [inline tox.ini:121]:
```suggestion
    'docs_dir = pathlib.Path(r"{envdir}") / "html"; index_file = docs_dir / "index.html"; print(f"\nDocumentation available at file://\{index_file\}\n\nTo serve docs, run: python3 -m http.server --directory \{docs_dir\} 0\n")'
```

File: docs/requirements.txt [hunk 10]
```diff
@@ -0,0 +1 @@
+Sphinx>=3.1.1
```
Comment (webknjaz) [inline docs/requirements.txt:1]: Unrelated to removing makefiles

**Comment by ssbarnea** [comment]
x

**Pull request closed without merging**